secboot/secboot.go

```diff
@@ -251,3 +253,56 @@ func MarkSuccessful() error {
 
 	return nil
 }
+
+// AuthMode corresponds to an authentication mechanism.
+type AuthMode string
+
+const (
+	AuthModePassphrase AuthMode = "passphrase"
+	AuthModePIN        AuthMode = "pin"
+)
+
+// VolumesAuthOptions contains options for the volumes authentication
+// mechanism (e.g. passphrase authentication).
+//
+// TODO: Add PIN option when secboot support lands.
+type VolumesAuthOptions struct {
```

Comment:
I think we might want to define this into gadget/device, maybe we should also move EncryptionType there, this would solve this in client:

        // Type has values of secboot.EncryptionType: "", "cryptsetup",
        // "cryptsetup-with-inline-crypto-engine"
        Type string `json:"encryption-type,omitempty"`

Reply:
notice that all relevant places import gadget/device already either directly or indirectly

Reply:
PR for `secboot.EncryptionType` https://github.com/canonical/snapd/pull/14855